### DISCUSSION TITLE
Procs parameter

### DISCUSSION THREAD

**Discussion by Carlos-Fernands** [MilesCranmer/PySR#153] [Q&A]
Hi,

I have 16 cores , but although I've been playing with the procs parameter I never seem to be using more than 1 core.
Is there anything else that needs to be done/setup/configured?

**Comment by MilesCranmer** [comment]
That seems like a bug... `procs` should be the number of cores you use. Are you using any other parameters different from the defaults?

**Comment by Carlos-Fernands** [comment]
My parameters are set in this code (a small variation of the simple example provided):

from pysr import PySRRegressor
model = PySRRegressor(
    procs = 8,
    batching = True,
    batch_size = 5000,
    model_selection="best",  # Result is mix of simplicity+accuracy
    niterations=40,
    binary_operators=["+", "*"],
    unary_operators=[
        "cos",
        "exp",
        "sin",
        "inv(x) = 1/x",
	# ^ Custom operator (julia syntax)
    ],
    extra_sympy_mappings={"inv": lambda x: 1 / x},
    # ^ Define operator for SymPy as well
    loss="loss(x, y) = (x - y)^2",
    # ^ Custom loss function (julia syntax)
)

**Comment by MilesCranmer** [comment]
Ah, so the `batch_size` is quite large. Could you try with a much smaller batch_size? Maybe just down-sample your data to ~1000 points, and see how well you can fit it with that? Generally symbolic regression is much less expressive than other ML techniques, so you don't need to worry as much about overfitting.

**Reply by MilesCranmer** [reply]
i.e., `batching=False`, then `X = X[:1000]; y=y[:1000]`, or equivalent

**Reply by Carlos-Fernands** [reply]
Did exactly as you suggested - same result, but with a difference in the output: this time there was none!
Same thing when I changed the 1000 to 2000 or to 3000
I then thought that it could have been because I ran a PCA to reduce the independent variables from 12 down to 6 and so I ran the code without applying the PCA, but still restricting the number of rows to 1000. Again no output.
Finally, I tried the select_k_features = 6 parameter - same result.
So, the problem is obviously elsewhere. I can't believe I need the many thousands of rows I used before to get an output (as I did).
I'll keep increasing the number of rows to see where it starts working. In any case, I never had more than 1 core working in all of these runs.

I may prepare a toy problem,  around some common data set, and put the code here, to see if it is reproducible.

**Reply by MilesCranmer** [reply]
Can you put this in an issue actually? Discussions is more for talking about best practices or ideas where your issue seems like an actual code issue.

When you make the issue, make sure to put the complete code you are using, after making sure you can run the example on the README.

**Comment by mtaylorDW** [comment]
@Carlos-Fernands, I had a similar problem.  Setting `multithreading=False` fixed it.  Do you have an AMD CPU by any chance?https://github.com/MilesCranmer/PySR/issues/158

**Reply by Carlos-Fernands** [reply]
No,  I have an Intel CPU

**Reply by Carlos-Fernands** [reply]
I should add that in my case the real problem was that an environment variable (JULIA_NUM_THREADS) was not set when Julia was installed, as part of the pySR installation. After setting "manually" this to my max, 16 , and a restart of the environment all was Ok
